WordPressAuthenticator/Unified Auth/View Related/Site Address/SiteAddressViewController.swift

```diff
@@ -416,8 +434,42 @@ private extension SiteAddressViewController {
 
         configureViewLoading(true)
 
+        guard let url = URL(string: loginFields.siteAddress) else {
+            configureViewLoading(false)
+            return displayError(message: Localization.invalidURL, moveVoiceOverFocus: true)
+        }
+
+        // Checks that the site exists
+        var request = URLRequest(url: url)
+        request.httpMethod = "HEAD"
+        request.timeoutInterval = 10.0 // waits for 10 seconds
+        let task = URLSession.shared.dataTask(with: request) { [weak self] _, _, error in
+            DispatchQueue.main.async { [weak self] in
+                guard let self = self else { return }
+
+                if let error = error {
+                    self.configureViewLoading(false)
+
+                    if self.authenticationDelegate.shouldHandleError(error) {
+                        self.authenticationDelegate.handleError(error) { customUI in
+                            self.pushCustomUI(customUI)
+                        }
+                        return
+                    }
+
+                    return self.displayError(message: Localization.nonExistentSiteError, moveVoiceOverFocus: true)
+                }
+
+                // Proceeds to check for the site's WordPress
+                self.guessXMLRPCURL(for: self.loginFields.siteAddress)
+            }
+        }
+        task.resume()
+    }
```

Comment:
❓ 
Should we raise a PR in the WP iOS repository and test this change as this code will execute even if `isSiteDiscovery` is false? 🤔 
Internal reference - p91TBi-49U-p2

Reply:
Makes sense! Added a PR here: https://github.com/wordpress-mobile/WordPress-iOS/pull/19209.